src/api/parser/test/indexer.test.js

```diff
@@ -0,0 +1,187 @@
+const { POSTS_URL = 'http://localhost/v1/posts' } = process.env;
+const { Elastic, logger, createError } = require('@senecacdot/satellite');
+const { waitOnReady, indexPost, deletePost, search } = require('../src/utils/indexer');
+
+const index = 'posts';
+const { mock } = Elastic();
+const loggerSpy = jest.spyOn(logger, 'error').mockImplementation((msg) => msg);
+
+afterEach(() => {
+  mock.clearAll();
+  jest.clearAllMocks();
+});
+
+const mockPost = {
+  id: 1111,
+  text: 'mock text 1',
+  title: 'mock title 1',
+  published: '2015-01-01',
+  author: 'mock author 1',
+};
+
+const differentPost = {
+  id: 2222,
+  text: 'mock text 2',
+  title: 'mock title 2',
+  published: '2015-01-01',
+  author: 'mock author 2',
+};
+
+describe('"Posts" index creation tests', () => {
+  const createSpy = jest.fn().mockReturnValue({});
+  it('Established connection and successful index creation should have no errors or logged errors', async () => {
+    mock.add(
+      {
+        method: 'PUT',
+        path: `/${index}`,
+      },
+      createSpy
+    );
+    await waitOnReady();
```

Comment:
Do this once in a `beforeEach()`

Reply:
I'm not sure which part you're referring to, is it for `await waitOnReady()` ? 
The two mocks in this describe are slightly different. I need to set up the mocks before calling `waitOnReady()`, cause `createPostsIndex()` is called in there.

Reply:
Ah, it seemed you were doing the same pattern multiple times, which is a "bad code smell" and often means you can refactor.  If you need to repeat, do it for sure.

Reply:
I've changed it so `mock.add()` is wrapped in a function that can be called for each unit test.